blog/macos-rig/index.qmd

```diff
@@ -0,0 +1,309 @@
+
+:::{.callout-tip}
+On a Mac, you can press Command + Shift + . to see your hidden folders.
+:::
+
+A `custom.zsh` file might be helpful because you can add many more commands for OMZ plugins than what are built in. For instance,
+
+```bash
+alias bog="brew outdated --greedy"
+```
+
+This allows you to use `bog` to run `brew outdated --greedy`, instead of having to type the whole command out. That particular command is helpful if you want to find ALL the programs that could potentially be upgraded.
+
+![](image1.jpg){fig-align="center" fig-alt="Keep Calm and bubo bog bupc bc" fig-width="400"}
+
+7. Go back to the Terminal and install [rig](https://github.com/r-lib/rig#linux).
+
+[rig](https://github.com/r-lib/rig#linux), the 'R Installation Manager,' is an amazing tool. It not only lets you download R, but install, configure, switch between, and remove _multiple_ R versions. It also allows you to manage _user-level package libraries_ (meaning you can delete a version of R without losing all your packages!). Mara Averick wrote a great summary on [Day 10 of Tidy Advent 2022](https://mara.quarto.pub/tidy-advent-2022/posts/10-rig.html).
+
+Install rig by running these commands in the Terminal:
+
+```bash
+brew tap r-lib/rig
+brew install --cask rig
+```
+
+8. Install R.
+
+Install the latest version of R using rig:
+
+```bash
+rig add release
+```
+
+9. Install [RStudio Desktop](https://posit.co/download/rstudio-desktop).
+
+You want the latest and greatest RStudio, no? Use Homebrew to download and install the latest version. Once a new version is released, the community is very quick at updating the cask entry for RStudio in Homebrew.
+
+```bash
+brew install --cask rstudio
+```
+
+10. <a name="abcd">Configure [`.Rprofile`](https://github.com/gvelasq/dotfiles/blob/main/R/.Rprofile).</a>
+
+In RStudio, install the {[usethis](https://usethis.r-lib.org/)} package, a tremendous tool for increased productivity with your R projects. In other words, you should use {usethis} 😀.
+
+```r
+install.packages(“usethis”)
+```
+
+Once installed, open your `.Rprofile` by running:
+
+```r
+usethis::edit_r_profile()
+```
+
+[Here's an example of an `.Rprofile` script.](https://github.com/gvelasq/dotfiles/blob/main/R/.Rprofile) This script [1] sets {devtools} and {usethis} to load with each R session (to write `install_github()` without a `library(devtools)`), [2] sets [RSPM](https://packagemanager.rstudio.com/client/#/) as the default repo for Linux (not relevant for this macOS guide), [3] sets a bunch of {usethis} defaults for package development which need to be overridden with personal details for each person, [4] sets `vsc.rstudioapi = TRUE` for VS Code (again not relevant for this guide), [5] sets several strict warnings which are helpful when developing packages (e.g., not using partial argument matching).
```

Comment:
Should RSPM now be PPPM? In which case perhaps the full acronym should be spelled out with PPPM in parens.